web/components/templates/requests/components/chatComponent/ChatMessage.tsx

```diff
@@ -246,7 +596,106 @@ export default function ChatMessage({
+  const deleteContentArrayItem = (contentIndex: number) => {
+    if (!onChatChange) return;
+
+    onChatChange({
+      ...mappedRequest,
+      schema: {
+        ...mappedRequest.schema,
+        request: {
+          ...mappedRequest.schema.request,
+          messages: mappedRequest.schema.request?.messages
+            ?.map((msg, i) => {
+              if (i === messageIndex && msg._type === "contentArray") {
+                const updatedContentArray = msg.contentArray?.filter(
+                  (_, j) => j !== contentIndex
+                );
+
+                if (updatedContentArray?.length === 1) {
+                  const remainingItem = updatedContentArray[0];
+                  if (remainingItem._type === "message") {
+                    return {
+                      ...remainingItem,
+                      role: msg.role,
+                      id: msg.id,
+                    };
+                  }
+                }
+
+                if (!updatedContentArray || updatedContentArray.length === 0) {
+                  return null;
+                }
+
+                return {
+                  ...msg,
+                  contentArray: updatedContentArray,
+                };
+              }
+              return msg;
+            })
+            .filter((msg): msg is Message => msg !== null),
+        },
+      },
+    });
+  };
+
+  const addTextToMessage = () => {
+    if (!onChatChange) return;
+
+    onChatChange({
+      ...mappedRequest,
+      schema: {
+        ...mappedRequest.schema,
+        request: {
+          ...mappedRequest.schema.request,
+          messages: mappedRequest.schema.request?.messages?.map((msg, i) => {
+            if (i === messageIndex) {
+              if (msg._type === "contentArray" && msg.contentArray) {
+                return {
+                  ...msg,
+                  contentArray: [
+                    ...msg.contentArray,
+                    {
+                      _type: "message" as const,
+                      role: "user",
+                      content: "",
+                      id: `text-${messageIndex}-${Date.now()}`,
+                    },
+                  ],
+                };
+              }
+
+              const contentArray = [];
+
+              contentArray.push({
+                _type: "message" as const,
+                role: "user",
+                content: msg.content,
+                id: `text-${messageIndex}-${Date.now()}`,
+              });
+
+              contentArray.push({
+                _type: "message" as const,
+                role: "user",
+                content: "",
+                id: `text-${messageIndex}-${Date.now() + 1}`,
+              });
+
+              return {
+                content: "",
+                _type: "contentArray" as const,
+                role: "user",
+                contentArray,
+                id: msg.id || `msg-${messageIndex}-${Date.now()}`,
+              };
```

Comment:
can probably shorten this by just having the objects inside the array while instantiating itself